### PR TITLE
added k8s deployment, mv Dockerfile to deployment directory

### PR DESCRIPTION
**Pull request by zhz2017**
These changes should not affect currently running Gateway.

### CODE REVIEW

File: deployment/sia-dev-gw-in-ppe-cluster.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: sia-dev-gw-svc
```
Comment (strand) [inline deployment/sia-dev-gw-in-ppe-cluster.yml]: Is it possible to rename to `sia-dev-gateway-service`? (I prefer longer, more descriptive names and variables.)

File: deployment/sia-dev-gw-in-ppe-cluster.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  - port: 80
+    name: http
+  selector:
+    run: sia-dev-gw
+    
+---
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: sia-dev-gw
```
Comment (strand) [inline deployment/sia-dev-gw-in-ppe-cluster.yml]: Can we replace every instance of ` sia-dev-gw` with  `sia-dev-gateway`?

File: deployment/sia-dev-gw-in-ppe-cluster.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      run: sia-dev-gw
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        run: sia-dev-gw
+    spec:
+      containers:
+          - name: sia-gateway-dev
+            image: siacontainerregistry.azurecr.io/azure/sia-gateway-dev:latest
```
Comment (strand) [inline deployment/sia-dev-gw-in-ppe-cluster.yml]: Please rename `sia-gateway-dev` to `sia-dev-gateway` to maintain consistency with other names starting with `sia-dev-`.

**Comment by zhz2017** [comment]
Thanks for the comments. I should clarify this deployment is for my exploring only. I will use more formal names as suggested for PPE and PROD stages whence we get VSTS Kubernetes deployment figured out.

**Comment by zhz2017** [comment]
Merge PR now so we can work out details in the VSTS task.

**Pull request merged**